AdyenActions/Actions/RedirectAction.swift

```diff
@@ -8,25 +8,64 @@ import Foundation
 
 /// Describes an action in which the user is redirected to a URL.
 public struct RedirectAction: Decodable {
-    
+
+    /// Defines the type of redirect flow utilized by the `RedirectAction` object.
+    public enum RedirectType: String, Decodable {
+        case redirect
+        case nativeRedirect
+
+        public init(from decoder: Decoder) throws {
+            let container = try decoder.singleValueContainer()
+            let type = try container.decode(String.self)
+            self = RedirectType(rawValue: type) ?? .redirect
+        }
+    }
+
     /// The URL to which to redirect the user.
     public let url: URL
-    
+
     /// The server-generated payment data that should be submitted to the `/payments/details` endpoint.
     public let paymentData: String?
-    
+
+    /// Redirect type.
+    public let type: RedirectType
```

Comment:
Leaving this comment to not forget the conversation we had regarding the backend behavior